portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,33 +28,89 @@
 @WebServlet("/comments")
 public class DataServlet extends HttpServlet {
 
-  private final List<Comment> sampleComments = new ArrayList<Comment>();
+  private final String UNSAFE_CHARACTERS_REGEX = "[^A-Za-z0-9._~()'!*:@,;+?\\s-]";
 
-  @Override
-  public void init() {
-    // Add sample data
-    sampleComments.add(new Comment(
-      "Richie Goulazian",
-      "This is an example of a comment. This is one of multiple comments that I need to write"
-    ));
-    sampleComments.add(new Comment(
-      "Rich Goulaz",
-      "This is not an example of a comment... or maybe it is? How to tell..."
-    ));
-    sampleComments.add(new Comment(
-      "Bob Guy",
-      "My name is Bob. I like this page."
-    ));
-  }
+  private final List<Comment> sessionComments = new ArrayList<Comment>();
+  private final Gson gson = new Gson();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Convert comments to JSON
     Gson gson = new Gson();
-    String commentJson = gson.toJson(sampleComments);
+    String commentJson = gson.toJson(sessionComments);
 
     // Send the JSON as the response
     response.setContentType("application/json;");
     response.getWriter().println(commentJson);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get new comment parameters
+    final String COMMENT_AUTHOR_PARAM = "author";
+    final String COMMENT_BODY_PARAM = "comment-body";
+    String commentAuthor = getParameter(request, COMMENT_AUTHOR_PARAM, "");
+    String commentBody = getParameter(request, COMMENT_BODY_PARAM, "");
```

Comment:
These are only used once and `COMMENT_AUTHOR_PARAM` doesn't tell me anything that I wouldn't get from reading `getParameter(request, "author", "");` In this case it is clear that "author" is the parameter that we're going to get from the request.

This same logic applies to COMMENT_BODY_PARAM.

```suggestion
    String commentAuthor = getParameter(request, "author", "");
    String commentBody = getParameter(request, "comment-body", "");
```

Reply:
Yeah that makes sense, committing your change.